### PR TITLE
SLING-11438 - Resource path consising of %7D with multiple dots leads…

### PR DESCRIPTION
**Pull request by sagarmiglani**
… to path traversal

### CODE REVIEW

**Review by karlpauls — APPROVED** [review]
LGTM

File: src/main/java/org/apache/sling/engine/impl/request/RequestData.java [hunk 1]
```diff
@@ -763,4 +768,34 @@ public SlingHttpServletRequest createRequest(RequestData requestData, HttpServle
     private static interface SlingHttpServletRequestFactory {
         SlingHttpServletRequest createRequest(RequestData requestData, HttpServletRequest request);
     }
+
+    /*
+    * Traverses the path segment wise and checks
+    * if there is any path with only dots (".")
+    * skipping SKIPPED_TRAVERSAL_CHARS characters in segment.
+    */
+    private static boolean travesesParentPath(String path) {
```
Comment (kwin) [inline src/main/java/org/apache/sling/engine/impl/request/RequestData.java:777]: small typo here: traversal

File: src/main/java/org/apache/sling/engine/impl/request/RequestData.java [hunk 2]
```diff
@@ -763,4 +767,34 @@ public SlingHttpServletRequest createRequest(RequestData requestData, HttpServle
...
     }
+
+    /*
+    * Traverses the path segment wise and checks
+    * if there is any path with only dots (".")
+    * skipping SKIPPED_TRAVERSAL_CHARS characters in segment.
+    */
+    private static boolean traversesParentPath(String path) {
+        int index = 0;
+        while(index < path.length()) {
```
Comment (kwin) [inline src/main/java/org/apache/sling/engine/impl/request/RequestData.java:778]: space after while missing

File: src/main/java/org/apache/sling/engine/impl/request/RequestData.java [hunk 3]
```diff
@@ -763,4 +767,34 @@ public SlingHttpServletRequest createRequest(RequestData requestData, HttpServle
...
+            int charCount = 0;
+            int dotCount = 0;
+            // count dots (".") and total chars in each path segment (between two '/')
+            while(index < path.length() && path.charAt(index) != '/') {
```
Comment (kwin) [inline src/main/java/org/apache/sling/engine/impl/request/RequestData.java:782]: same here

File: src/main/java/org/apache/sling/engine/impl/request/RequestData.java [hunk 4]
```diff
@@ -763,4 +767,34 @@ public SlingHttpServletRequest createRequest(RequestData requestData, HttpServle
...
+                char c = path.charAt(index);
+                if (!SKIPPED_TRAVERSAL_CHARS.contains(c)) {
+                    if (c == '.') dotCount++;
```
Comment (kwin) [inline src/main/java/org/apache/sling/engine/impl/request/RequestData.java:785]: always add brackets

**Review by kwin — APPROVED** [review]
just some styling remarks, apart from that LGTM

File: src/main/java/org/apache/sling/engine/impl/request/RequestData.java (hunk 1 above)
Comment (sagarmiglani) [reply src/main/java/org/apache/sling/engine/impl/request/RequestData.java]: Fixed

**Comment by sagarmiglani** [comment]
Thanks @kwin for the comments, updated the code.
@kwin @karlpauls if all looks good, can we merge and release? Thanks!

**Pull request merged**